### PR TITLE
Adding a Contributor Ladder

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This adds a contributor ladder. This nearly identical to the [original doc](https://docs.google.com/document/d/19l77HW0jfOOu_ckMqu16-CC6W3GrQDiO8lmU9cj9LDc/edit#heading=h.s2dc1trxu6gv).

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/hold for consensus

### CODE REVIEW

File: site-src/contributing/contributor-ladder.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+The first steps on the contributor ladder are already [clearly defined in the
+upstream Kubernetes
+Community](https://github.com/kubernetes/community/blob/master/community-membership.md#community-membership).
+Gateway API follows those guidelines along with the rest of the Kubernetes
+community. Within Gateway API, there are a variety of areas one can become a
+reviewer or approver, this includes:
+
+* Conformance
+* Documentation
+* GEPs
```
Comment (gyohuangxin) [inline site-src/contributing/contributor-ladder.md:87]: Can `Webhook Validation` be one of these areas? IMO, it also has code to be reviewed.

**Comment by youngnick** [comment]
LGTM as is, although I think that @gyohuangxin's idea about webhook validation as a separate area is also a good one.

File: site-src/contributing/contributor-ladder.md (hunk 1 above)
Comment (robscott) [reply site-src/contributing/contributor-ladder.md:87]: Good call, added, thanks!

**Comment by robscott** [comment]
Thanks for the feedback! Updated to add webhook validation as an example, PTAL.

**Review by shaneutt — APPROVED** [review]

**Review by gyohuangxin — APPROVED** [review]

**Review by Xunzhuo — APPROVED** [review]
LGTM

**Comment by robscott** [comment]
Think we've got consensus, removing hold. @shaneutt @youngnick @Xunzhuo can one of you LGTM this?

**Review by shaneutt — APPROVED** [review]
/lgtm

**Pull request merged**